### PR TITLE
[TD-2705] Pre chat lead collection

### PR DESCRIPTION
**Pull request by pranavp99**
### What do you want to achieve?
-Pass Pre chat lead collection data to Zendesk

### PR Checklist

- [x] I have tested it locally and all functionalities are working fine.
- [ ] I have compared it with mocks and all design elements are the same.
- [ ] I have tested it in IE Browser.

### How was the code tested?

-Ran the code locally to check if data is being passed to Zendesk

### What new thing you came across while writing this code? 
-

### In case you fixed a bug then please describe the root cause of it? 
-

NOTE: Make sure you're comparing your branch with the correct base branch

### CODE REVIEW

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 1]
```diff
@@ -2288,6 +2289,7 @@ var userOverride = {
                 }
             };
             _this.initialize = function (userPxy) {
+                window.console.log("checker!");
```
Comment (gargshantnu) [inline webplugin/js/app/mck-sidebox-1.0.js:2292]: remove debug statements

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 2]
```diff
@@ -2660,7 +2664,7 @@ var userOverride = {
                 
                 // Loading zopim sdk for zendesk chat integration
                 if (kommunicate._globals.zendeskChatSdkKey) {
-                    zendeskChatService.init(kommunicate._globals.zendeskChatSdkKey);
+                    zendeskChatService.init(kommunicate._globals.zendeskChatSdkKey, preChatData);
```
Comment (gargshantnu) [inline webplugin/js/app/mck-sidebox-1.0.js:2667]: result must be available at this level. use that. no need to create a new variable `preChatData`

File: webplugin/js/app/zendesk-chat-service.js [hunk 3]
```diff
@@ -67,7 +68,7 @@ function ZendeskChatService() {
                     }
                 }
             }
-            zChat.init(zendeskInitOptions);
+            zChat.init(zendeskInitOptions, preChatData);
```
Comment (gargshantnu) [inline webplugin/js/app/zendesk-chat-service.js:71]: is this neeeded?

File: webplugin/js/app/zendesk-chat-service.js [hunk 4]
```diff
@@ -6,9 +6,9 @@ function ZendeskChatService() {
     var ZENDESK_SDK_INITIALIZED = false;
     var ZENDESK_CHAT_SDK_KEY = "";
 
-    _this.init = function (zendeskChatSdkKey) {
+    _this.init = function (zendeskChatSdkKey, preChatData) {
         ZENDESK_CHAT_SDK_KEY = zendeskChatSdkKey;
-        _this.loadZopimSDK();
+        _this.loadZopimSDK(preChatData);
```
Comment (gargshantnu) [inline webplugin/js/app/zendesk-chat-service.js:11]: see if we can store these user details on class level and then use it from there.
no need to pass on each level to all the middle functions

File: webplugin/js/app/mck-sidebox-1.0.js (hunk 2 above)
Comment (pranavp99) [reply webplugin/js/app/mck-sidebox-1.0.js:2665]: its not available at this point which is why i had to declare a global variable

File: webplugin/js/app/zendesk-chat-service.js [hunk 5]
```diff
@@ -5,9 +5,11 @@ function ZendeskChatService() {
     var _this = this;
     var ZENDESK_SDK_INITIALIZED = false;
     var ZENDESK_CHAT_SDK_KEY = "";
+    var preChatLeadData = "";
```
Comment (gargshantnu) [inline webplugin/js/app/zendesk-chat-service.js:8]: when assigning something to it, are we assigning Object or String?
we need to keep base value same as the final value in terms of data type

File: webplugin/js/app/mck-sidebox-1.0.js (hunk 2 above)
Comment (gargshantnu) [reply webplugin/js/app/mck-sidebox-1.0.js:2665]: can you check what all parameters are coming in this? and if we can use them
https://github.com/Kommunicate-io/Kommunicate-Web-SDK/blob/fd8e1833cf38a12db5c6e236a6b5a598c0bd2d1b/webplugin/js/app/mck-sidebox-1.0.js#L2428

File: webplugin/js/app/zendesk-chat-service.js [hunk 6]
```diff
@@ -36,9 +38,10 @@ function ZendeskChatService() {
             var zendeskInitOptions = {
                 account_key: ZENDESK_CHAT_SDK_KEY,
             }
-            var name = kommunicate._globals.name || kommunicate._globals.userName;
-            var email = kommunicate._globals.email;
-            var externalId = kommunicate._globals.userId;
+            //var initializeCallData;
```
Comment (gargshantnu) [inline webplugin/js/app/zendesk-chat-service.js:41]: remove commented code

**Comment by gargshantnu** [comment]
also if you have tested it, please tick the check box in PR description

**Review by gargshantnu — DISMISSED** [review]

**Review by mukeshthawani — DISMISSED** [review]

**Review by Prochnost55 — APPROVED** [review]

File: webplugin/js/app/zendesk-chat-service.js [hunk 7]
```diff
@@ -196,11 +199,20 @@ function ZendeskChatService() {
...
 
         console.log("handleZendeskAgentMessageEvent ", event);
+        var agentId = event.nick.replace(":", "-")
+        if (!AGENT_INFO_MAP[agentId]) {
+            AGENT_INFO_MAP[agentId] = {
+                displayName: event.display_name,
+                agentId: agentId
+            }
+        }
+        console.log("AGENT_INFO_MAP", AGENT_INFO_MAP);
```
Comment (gargshantnu) [inline webplugin/js/app/zendesk-chat-service.js:209]: did you manually copied the changes?

**Review by gargshantnu — APPROVED** [review]

File: webplugin/js/app/zendesk-chat-service.js (hunk 7 above)
Comment (gargshantnu) [reply webplugin/js/app/zendesk-chat-service.js:209]: talked over slack. some rebase issue, suggested to install github desktop

**Pull request merged**